### PR TITLE
Update command to install dependencies and run training

### PR DESCRIPTION
**Pull request by Chakradhar886**
# Description

# Checklist

- [x] I have read the contribution guidelines.
  - [General](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
  - [SDK](https://github.com/Azure/azureml-examples/blob/main/sdk/CONTRIBUTING.md)
  - [CLI](https://github.com/Azure/azureml-examples/blob/main/cli/CONTRIBUTING.md)
- [x] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [x] Pull request includes test coverage for the included changes.
- [x] This notebook or file is added to the [CODEOWNERS](https://github.com/Azure/azureml-examples/blob/main/.github/CODEOWNERS) file, pointing to the author or the author's team.

### CODE REVIEW

File: cli/jobs/pipelines/tensorflow-image-segmentation/pipeline.yml [hunk 1]
```diff
@@ -100,6 +100,7 @@ jobs:
             type: uri_folder
 
     command: >-
+        pip install 'protobuf<5' &&
```
Comment (Chakradhar886) [inline cli/jobs/pipelines/tensorflow-image-segmentation/pipeline.yml:103]: The tensorflow-image-segmentation pipeline job was failing because the curated environment (tensorflow-2.16-cuda12/labels/latest) received an upstream update that upgraded protobuf from 4.x to 5.x. Protobuf 5.x removed deprecated APIs that TensorFlow 2.16 still depends on, causing runtime crashes. This was previously passing because older builds of the curated environment included protobuf 4.x. Fixed by adding pip install 'protobuf<5' in the pipeline command to pin protobuf to a compatible version before running the training script.

Comment (Chakradhar886) [reply cli/jobs/pipelines/tensorflow-image-segmentation/pipeline.yml:103]: What are the changes required to support protobuf 5.x? @kshitij-microsoft 

To support protobuf 5.x, update ~15 files across the repo: bump all protobuf==3.19.x/3.20.x pins in conda YAMLs and requirements files to >=5.0,<6, upgrade the paired TensorFlow versions to ≥2.15 (since old TF is incompatible with protobuf 5.x), remove explicit protobuf<5 caps in pipeline commands, update inline extra_pip_requirements in Python source files, revise the setup.sh workaround for the 5.29.0 bug, and clean up the deprecated google.protobuf.service warning suppression in the test infrastructure since that module was removed in protobuf 4.0+.

**Review by achauhan-scc — APPROVED** [review]

**Review by kingernupur — APPROVED** [review]
